packages/mobile/src/pages/chat/chat-info-page.tsx

```diff
@@ -57,70 +53,123 @@ export const ChatInfoPage = memo((prop: ChatInfoProp) => {
     [isDarkMode, prop.navigation.goBack]
   );
 
-  // const headerRight = useCallback(
-  //   () =>
-  //     conversationContent && 'version' in conversationContent && conversationContent.version ? (
-  //       <TextButton
-  //         title="Edit"
-  //         unFilledStyle={{ marginRight: 8 }}
-  //         onPress={() => {
-  //           prop.navigation.navigate('EditGroup', {
-  //             convoId: conversationId,
-  //           });
-  //         }}
-  //       />
-  //     ) : null,
-  //   [conversationContent, conversationId, prop.navigation]
-  // );
+  const headerRight = useCallback(
+    () =>
+      conversationContent && 'version' in conversationContent && conversationContent.version ? (
+        <TextButton
+          title="Edit"
+          unFilledStyle={{ marginRight: 8 }}
+          onPress={() => {
+            prop.navigation.navigate('EditGroup', {
+              convoId: conversationId,
+            });
+          }}
+        />
+      ) : null,
+    [conversationContent, conversationId, prop.navigation]
+  );
 
-  const colorStyle = useMemo(() => {
-    return {
-      color: isDarkMode ? Colors.white : Colors.black,
-    };
-  }, [isDarkMode]);
   const headerStyle = useMemo(() => {
     return {
       backgroundColor: isDarkMode ? Colors.gray[900] : Colors.slate[50],
     };
   }, [isDarkMode]);
   if (!conversation) return null;
-  const hasGroupImage = conversation.fileMetadata.payloads?.length > 0;
-  const groupAvatarPayloadKey = conversation.fileMetadata.payloads?.[0]?.key;
+
+  const data = [...(recipients as string[]), identity as string] as string[];
+  const isAdmin = conversation.fileMetadata.senderOdinId === identity;
+
   return (
-    <>
+    <View
+      style={{
+        flex: 1,
+      }}
+    >
       <Header
         title={isGroup ? 'Group Info' : 'Chat Info'}
         headerLeft={headerLeft}
-        // headerRight={isGroup ? headerRight : undefined} //TODO: Enable this when we have figured out editing group conversations
+        headerRight={isGroup && isAdmin ? headerRight : undefined}
         headerStyle={headerStyle}
       />
-      <SafeAreaView>
-        <View style={styles.content}>
-          {!isGroup ? (
-            withYourself ? (
-              <OwnerAvatar style={styles.avatar} imageSize={styles.largeAvatarSize} />
-            ) : (
-              <Avatar
-                odinId={recipient as string}
-                style={styles.avatar}
-                imageSize={styles.largeAvatarSize}
-              />
-            )
+
+      <FlatList
+        data={data}
+        ListHeaderComponent={
+          <RenderListHeader
+            isGroup={isGroup || false}
+            recipient={recipient}
+            conversation={conversation}
+          />
+        }
+        keyExtractor={(item) => item}
```

Comment:
Best to move the keyExtractor and the renderItem into a useCallback wrapper to keep things performant